--- a/1.10.2/src/main/java/com/rwtema/extrautils2/utils/helpers/PlayerHelper.java
+++ b/1.10.2/src/main/java/com/rwtema/extrautils2/utils/helpers/PlayerHelper.java
@@ -1,6 +1,7 @@
 package com.rwtema.extrautils2.utils.helpers;
 
 import com.google.common.collect.Lists;
+import com.kanzaji.xu2patcher.XU2PatcherConfig;
 import com.mojang.authlib.GameProfile;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
@@ -39,6 +40,7 @@
 	}
 
 	private static boolean isTema(String name, UUID id) {
+		if (XU2PatcherConfig.get().cheatyTema) return false;
 		return "RWTema".equals(name) && id.equals(temaID);
 	}
 
