--- a/1.10.2/build.gradle
+++ b/1.10.2/build.gradle
@@ -1,7 +1,7 @@
 buildscript {
     repositories {
-        jcenter()
-        maven { url = "http://files.minecraftforge.net/maven" }
+        mavenCentral()
+        maven { url = "https://files.minecraftforge.net/maven" }
         maven {
             name = "sonatype"
             url = "https://oss.sonatype.org/content/repositories/snapshots/"
@@ -18,12 +18,12 @@
 
 apply plugin: 'net.minecraftforge.gradle.forge'
 
-ext.priv = parseConfig(file('../1.10.2/private.properties'))
+ext.priv = parseConfig(file('../1.10.2/version.properties'))
 ext.version_config = parseConfig(file('../1.10.2/version.properties'))
 
 version = ext.version_config.version_no
 
-group= "com.rwtema.extrautils2" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
+group= "com.rwtema.extrautils2" // https://maven.apache.org/guides/mini/guide-naming-conventions.html
 archivesBaseName = "extrautils2-1.10.2"
 
 
@@ -46,7 +46,7 @@
 
 repositories {
   maven {
-    url "http://dvs1.progwml6.com/files/maven"
+    url "https://dvs1.progwml6.com/files/maven"
   }
 }
 
