--- a/com/rwtema/extrautils2/backend/entries/XU2Entries.java
+++ b/com/rwtema/extrautils2/backend/entries/XU2Entries.java
@@ -65,9 +65,10 @@
 import net.minecraft.inventory.ContainerWorkbench;
 import net.minecraft.item.EnumDyeColor;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
 import net.minecraft.potion.PotionEffect;
+import net.minecraft.potion.PotionHelper;
 import net.minecraft.potion.PotionType;
-import net.minecraft.potion.PotionUtils;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.ResourceLocation;
@@ -78,8 +79,6 @@
 import net.minecraft.world.biome.Biome;
 import net.minecraft.world.chunk.Chunk;
 import net.minecraftforge.common.MinecraftForge;
-import net.minecraftforge.common.brewing.BrewingOreRecipe;
-import net.minecraftforge.common.brewing.BrewingRecipeRegistry;
 import net.minecraftforge.common.config.Configuration;
 import net.minecraftforge.oredict.OreDictionary;
 import org.apache.commons.lang3.tuple.Pair;
@@ -933,16 +932,10 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType antiFlying = PotionsHelper.registerPotionType(new PotionEffect(value, 60 * 20));
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getAwkwardPotionType()),
-					"obsidian",
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), antiFlying));
+			PotionHelper.func_193356_a(PotionsHelper.getAwkwardPotionType(), Ingredient.func_193369_a(OreDictionary.getOres("obsidian").toArray(new ItemStack[0])), antiFlying);
 
 			PotionType antiFlying2 = PotionsHelper.registerDerivedPotionType(new PotionEffect(value, 8 * 60 * 20), antiFlying, ".long");
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), antiFlying),
-					"dustRedstone",
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), antiFlying2));
+			PotionHelper.func_193356_a(antiFlying, Ingredient.func_193369_a(OreDictionary.getOres("dustRedstone").toArray(new ItemStack[0])), antiFlying2);
 		}
 	};
 	public static PotionEntry<PotionSecondChance> potionAntiDeath = new PotionEntry<PotionSecondChance>("potion_second_chance", "Will save from death but only once.") {
@@ -954,10 +947,7 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType antiDeath = PotionsHelper.registerPotionType(new PotionEffect(value, 2 * 60 * 20));
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getVanillaType("strong_healing")),
-					new ItemStack(Items.field_151153_ao, 1, 1),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), antiDeath));
+			PotionHelper.func_193356_a(PotionsHelper.getVanillaType("strong_healing"), Ingredient.func_193369_a(new ItemStack(Items.field_151153_ao, 1, 1)), antiDeath);
 		}
 	};
 	public static PotionEntry<PotionNapalm> potionGreekFire = new PotionEntry<PotionNapalm>("potion_greek_fire", "Creates a fire that cannot be extenguished") {
@@ -965,14 +955,8 @@
 		public void registerTypesAndRecipes() {
 			PotionType greekFire = PotionsHelper.registerPotionType(new PotionEffect(value, 2 * 60 * 20));
 			PotionType oily = PotionsHelper.newGenericPotion("Oily");
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getAwkwardPotionType()),
-					new ItemStack(Items.field_185164_cV),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), oily));
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), oily),
-					new ItemStack(Items.field_151129_at),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), greekFire));
+			PotionHelper.func_193356_a(PotionsHelper.getAwkwardPotionType(), Ingredient.func_193367_a(Items.field_185164_cV), oily);
+			PotionHelper.func_193356_a(oily, Ingredient.func_193367_a(Items.field_151129_at), greekFire);
 		}
 
 		@Override
@@ -984,11 +968,7 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType greekFire = PotionsHelper.registerPotionType(new PotionEffect(value, 30 * 20));
-
-			BrewingRecipeRegistry.addRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionTypes.field_185238_j),
-					new ItemStack(Items.field_151102_aT),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), greekFire));
+			PotionHelper.func_193356_a(PotionTypes.field_185238_j, Ingredient.func_193367_a(Items.field_151102_aT), greekFire);
 		}
 
 		@Override
@@ -1000,12 +980,7 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType relapse = PotionsHelper.registerPotionType(new PotionEffect(value, 8 * 60 * 20));
-			BrewingRecipeRegistry.addRecipe(new BrewingOreRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getAwkwardPotionType()),
-					Lists.newArrayList(
-							new ItemStack(Blocks.field_150428_aP)
-					),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), relapse)));
+			PotionHelper.func_193356_a(PotionsHelper.getAwkwardPotionType(), Ingredient.func_193369_a(new ItemStack(Blocks.field_150428_aP)), relapse);
 		}
 
 		@Override
@@ -1022,14 +997,13 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType love = PotionsHelper.registerPotionType(new PotionEffect(value, 10 * 20));
-			BrewingRecipeRegistry.addRecipe(new BrewingOreRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getAwkwardPotionType()),
-					Lists.newArrayList(
-							new ItemStack(Blocks.field_150398_cm, 1, BlockDoublePlant.EnumPlantType.ROSE.func_176936_a()),
-							new ItemStack(BlockFlower.EnumFlowerType.POPPY.func_176964_a().func_180346_a(), 1, BlockFlower.EnumFlowerType.POPPY.func_176968_b()),
-							new ItemStack(BlockFlower.EnumFlowerType.RED_TULIP.func_176964_a().func_180346_a(), 1, BlockFlower.EnumFlowerType.RED_TULIP.func_176968_b())
+			PotionHelper.func_193356_a(PotionsHelper.getAwkwardPotionType(),
+					Ingredient.func_193369_a(
+                            new ItemStack(Blocks.field_150398_cm, 1, BlockDoublePlant.EnumPlantType.ROSE.func_176936_a()),
+                            new ItemStack(BlockFlower.EnumFlowerType.POPPY.func_176964_a().func_180346_a(), 1, BlockFlower.EnumFlowerType.POPPY.func_176968_b()),
+                            new ItemStack(BlockFlower.EnumFlowerType.RED_TULIP.func_176964_a().func_180346_a(), 1, BlockFlower.EnumFlowerType.RED_TULIP.func_176968_b())
 					),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), love)));
+					love);
 		}
 	};
 	public static PotionEntry<PotionPurge> potionPurge = new PotionEntry<PotionPurge>("potion_Purge", null) {
@@ -1041,12 +1015,7 @@
 		@Override
 		public void registerTypesAndRecipes() {
 			PotionType Purge = PotionsHelper.registerPotionType(new PotionEffect(value));
-			BrewingRecipeRegistry.addRecipe(new BrewingOreRecipe(
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), PotionsHelper.getAwkwardPotionType()),
-					Lists.newArrayList(
-							new ItemStack(Items.field_151078_bh)
-					),
-					PotionUtils.func_185188_a(new ItemStack(Items.field_151068_bn), Purge)));
+			PotionHelper.func_193356_a(PotionsHelper.getAwkwardPotionType(), Ingredient.func_193367_a(Items.field_151078_bh), Purge);
 		}
 	};
 	public static VillagerEntrySimple alchemist = new VillagerEntrySimple("alchemist") {
