--- a/com/rwtema/extrautils2/XU2Entries112.java
+++ b/com/rwtema/extrautils2/XU2Entries112.java
@@ -17,6 +17,7 @@
 	public static final ItemClassEntry<ItemLuxSaber> itemLuxSaber = new ItemClassEntry<ItemLuxSaber>(ItemLuxSaber.class) {
 		@Override
 		public void addRecipes() {
+			System.out.println("TESTESTESTSET");
 
 			ItemStack crystal = XU2Entries.sunCrystal.newStack(1, 0);
 			for (int i = 0; i < LuxColors.values().length; i++) {
