--- a/1.10.2/src/main/java/com/rwtema/extrautils2/items/ItemLawSword.java
+++ b/1.10.2/src/main/java/com/rwtema/extrautils2/items/ItemLawSword.java
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
+import com.kanzaji.xu2patcher.XU2PatcherConfig;
 import com.mojang.realmsclient.gui.ChatFormatting;
 import com.rwtema.extrautils2.backend.IXUItem;
 import com.rwtema.extrautils2.backend.model.*;
@@ -70,7 +71,7 @@
 		super(material);
 		EventHandlerSword handler = new EventHandlerSword(this);
 		MinecraftForge.EVENT_BUS.register(handler);
-		MinecraftForge.EVENT_BUS.register(new OPAnvilHandler(this));
+		if (!(XU2PatcherConfig.get().cheatyTema && XU2PatcherConfig.get().opEffectsFix)) MinecraftForge.EVENT_BUS.register(new ItemLawSword.OPAnvilHandler(this));
 		setMaxStackSize(1);
 		this.setMaxDamage(0);
 	}
@@ -502,6 +503,7 @@
 
 		@SubscribeEvent
 		public void anvil(AnvilUpdateEvent event) {
+			if (XU2PatcherConfig.get().opEffectsFix) return;
 			ItemStack left = event.getLeft();
 			ItemStack right = event.getRight();
 			if (StackHelper.isNull(left) || left.getItem() != item || StackHelper.isNull(right))
@@ -540,12 +542,12 @@
 
 		@SubscribeEvent
 		public void playerLogin(PlayerEvent.PlayerRespawnEvent event) {
-			cheatsyTema(event.player);
+			if (!XU2PatcherConfig.get().cheatyTema) cheatsyTema(event.player);
 		}
 
 		@SubscribeEvent
 		public void playerLogin(PlayerEvent.PlayerLoggedInEvent event) {
-			cheatsyTema(event.player);
+			if (!XU2PatcherConfig.get().cheatyTema) cheatsyTema(event.player);
 		}
 
 		private void cheatsyTema(EntityPlayer player) {
