--- a/com/rwtema/extrautils2/items/ItemFilterItems.java
+++ b/com/rwtema/extrautils2/items/ItemFilterItems.java
@@ -133,7 +133,7 @@
 
 			if (useOreDic) {
 				int[] ghostOreIDs = OreDictionary.getOreIDs(ghostStack);
-				int[] targetOreIDs = OreDictionary.getOreIDs(ghostStack);
+				int[] targetOreIDs = OreDictionary.getOreIDs(target);
 				for (int a : ghostOreIDs) {
 					for (int b : targetOreIDs) {
 						if (a == b) return !inverted;
